include/swift/Remote/MemoryReader.h

```diff
@@ -134,10 +134,21 @@ class MemoryReader {
     // Default implementation returns the read value as is.
     return RemoteAbsolutePointer("", readValue);
   }
-  
+
+  /// Atempt to resolve the pointer to a symbol for the given remote address.
+  virtual llvm::Optional<RemoteAbsolutePointer>
+  resolvePointerAsSymbol(RemoteAddress address) {
```

Comment:
@adrian-prantl was your suggestion that we refactor the existing one? As is, this would give two functions that do the same.

Reply:
This variant works fine.

Reply:
Adrian do you mean we should add this new function and remove the old function?

The way I see it, the API doesn't need both `resolvePointer` and `resolvePointerAsSymbol`. I think this new one is the right way to model this behavior, and I think we can and should remove `resolvePointer`.

I also think we should attempt to give `resolvePointerAsSymbol` a better name. There's already `getSymbolAddress`, and as this function is the inverse, maybe it should be named `getAddressSymbol`. I think `getSymbolForAddress` might be most clear, but it would be confusing to have both `getSymbolAddress` and `getSymbolForAddress`. Maybe `symbolForAddress`?